### PR TITLE
Remove XIR files and clean up documentation

### PR DESCRIPTION
**Pull request by Mandrenkov**
**Context:**
The XIR is now available on [PyPI](https://pypi.org/project/quantum-xir/) and is being actively maintained in a [different repository](https://github.com/XanaduAI/xir).

After this PR is merged, Jet can be released on PyPI.

**Description of the Change:**
- Removed all XIR source files and tests (the former is now fetched from PyPI).
- Updated the XIR interpreter to use the latest version and conventions of the XIR.
- Created a standalone `requirements.txt` file as per the [PyPA guidelines](https://packaging.python.org/discussions/install-requires-vs-requirements/).
- Renamed `get_xir_library()` to `get_xir_manifest()` and added declarations for the supported outputs.
- Fixed a number of small issues including typos, missing hyperlinks, and outdated documentation.

**Benefits:**
- Jet is more streamlined and no longer contains any XIR implementation or tests.
- The high-level documentation is more current and consistent with other Xanadu repositories.

**Possible Drawbacks:**
- Some existing XIR scripts may no longer be interpretable by Jet since the previous revision of the XIR interpreter was based on an old version of the XIR.

**Related GitHub Issues:**
None.

### CODE REVIEW

**Review by trbromley — COMMENTED** [review]
Thanks @Mandrenkov! I just had a few questions about the links in the readme

File: README.rst [hunk 1]
```diff
@@ -157,12 +157,15 @@ Contributing to Jet
 We welcome new contributions - simply fork the Jet repository and make a
 `pull request <https://help.github.com/articles/about-pull-requests/>`_
 containing your contribution.  All contributers to Jet will be listed as authors
-on the releases.  See our `changelog <.github/CHANGELOG.md>`_ for more details.
+on the releases.  See our `changelog
+<https://raw.githubusercontent.com/XanaduAI/jet/.github/CHANGELOG.md>`_ for more
+details.
 
 We also encourage bug reports, suggestions for new features and enhancements,
 and even links to cool projects or applications built using Jet.  Visit the
```
Comment (thisac) [inline README.rst:165]: Do you follow the 2 spaces convention, or is this a typo?
```suggestion
and even links to cool projects or applications built using Jet. Visit the
```

File: python/requirements.txt [hunk 2]
```diff
@@ -1,6 +1,5 @@
-black
-cmake>=3.14
-isort>5
-pylint
-pytest>=5,<6
-wheel
+lark-parser==0.12.0
```
Comment (thisac) [inline python/requirements.txt:1]: Should `lark-parser` actually be a requirement here? Isn't that a requirement only for `quantum-xir` (making it only indirectly a requirement for Jet)?

File: python/jet/interpreter.py [hunk 3]
```diff
@@ -43,43 +41,48 @@ def get_xir_library() -> XIRProgram:
...
+def run_xir_program(program: xir.Program) -> List[Union[np.number, np.ndarray]]:
     """Executes an XIR program.
 
     Raises:
         ValueError: If the given program contains an unsupported gate or output
             statement or an invalid gate definition.
 
     Args:
-        program (xir.program.XIRProgram): XIR script to execute.
+        program (xir.Program): XIR script to execute.
```
Comment (thisac) [inline python/jet/interpreter.py:85]: This isn't really a script. A `program` or a `circuit` might be more correct.
```suggestion
        program (xir.Program): XIR program to execute.
```

File: python/tests/test_interpreter.py [hunk 4]
```diff
@@ -22,19 +23,17 @@ class TestGetXIRLibrary:
...
-                    end;
-
-                    gate Z[0]:
-                        Z | [0];
-                    end;
+                    gate X[0];
+                    gate Y[0];
+                    gate Z[0];
+                    out Amplitude;
+                    out Probabilities;
```
Comment (thisac) [inline python/tests/test_interpreter.py:36]: Nice! 🎸

File: python/tests/test_interpreter.py [hunk 5]
```diff
@@ -85,109 +73,116 @@ class TestGetXIRLibrary:
...
-            "gate t[0]: t | [0]; end; "
-            "gate toffoli[0, 1, 2]: toffoli | [0, 1, 2]; end; "
-            "gate twomodesqueezing(r, theta)[0, 1]: twomodesqueezing(r, theta) | [0, 1]; end; "
-            "gate u1(phi)[0]: u1(phi) | [0]; end; "
-            "gate u2(phi, lam)[0]: u2(phi, lam) | [0]; end; "
-            "gate u3(theta, phi, lam)[0]: u3(theta, phi, lam) | [0]; end; "
-            "gate x[0]: x | [0]; end; "
-            "gate y[0]: y | [0]; end; "
-            "gate z[0]: z | [0]; end;"
+            "gate BS(theta, phi)[0, 1]; "
```
Comment (thisac) [inline python/tests/test_interpreter.py:102]: Very nice! 🚀

**Review by thisac — COMMENTED** [review]
Great work @Mandrenkov! 💯 Nothing major blocking, will approve as soon as the comments from myself and Tom are addressed. 🙂

File: .github/CONTRIBUTING.md [hunk 6]
```diff
@@ -67,7 +67,7 @@ To submit a bug report, please work your way through the following checklist:
 
 * **Fill out the issue template**. If you cannot find an existing issue
   addressing the problem, create a new issue by filling out the
-  [bug issue template](./ISSUE_TEMPLATE/BUG.md). This template is added
+  [bug issue template](./ISSUE_TEMPLATE/bug-report.md). This template is added
```
[the remark this answers is not in the document]
Comment (Mandrenkov) [reply .github/CONTRIBUTING.md:70]: Great catch! The link you provided is definitely more useful. 🔗

File: README.rst [hunk 7]
```diff
@@ -1,7 +1,7 @@
 .. raw:: html
 
     <p align="center">
-        <img width="250" alt="Jet" src="docs/_static/jet_title.svg">
+        <img width="250" alt="Jet" src="https://raw.githubusercontent.com/XanaduAI/jet/docs/_static/jet_title.svg">
```
[the remark this answers is not in the document]
Comment (Mandrenkov) [reply README.rst:4]: Oops! 🤦‍♂️ Thanks for catching this.

File: README.rst [hunk 8]
```diff
@@ -157,12 +157,15 @@ Contributing to Jet
 We welcome new contributions - simply fork the Jet repository and make a
 `pull request <https://help.github.com/articles/about-pull-requests/>`_
 containing your contribution.  All contributers to Jet will be listed as authors
-on the releases.  See our `changelog <.github/CHANGELOG.md>`_ for more details.
+on the releases.  See our `changelog
+<https://raw.githubusercontent.com/XanaduAI/jet/.github/CHANGELOG.md>`_ for more
```
[the remark this answers is not in the document]
Comment (Mandrenkov) [reply README.rst:161]: Excellent catch! The raw content is necessary for images but should not be used for Markdown files!

File: README.rst [hunk 9]
```diff
@@ -157,12 +157,15 @@ Contributing to Jet
...
+details.
 
 We also encourage bug reports, suggestions for new features and enhancements,
 and even links to cool projects or applications built using Jet.  Visit the
-`contributions page <.github/CONTRIBUTIONS.md>`_ to learn more about sharing
-your ideas with the Jet team.
+`contributions page
+<https://raw.githubusercontent.com/XanaduAI/jet/.github/CONTRIBUTIONS.md>`_ to
```
[the remark this answers is not in the document]
Comment (Mandrenkov) [reply README.rst:167]: Yep, this should also be changed.

File: README.rst (hunk 1 above)
Comment (Mandrenkov) [reply README.rst:165]: Sorry, this is just an old habit of mine. Fixed!

File: python/requirements.txt (hunk 2 above)
Comment (Mandrenkov) [reply python/requirements.txt]: Nope, thanks for catching this! 🥅

File: python/jet/interpreter.py [hunk 10]
```diff
@@ -43,43 +41,48 @@ def get_xir_library() -> XIRProgram:
...
-        lines.append(line)
+    return program
 
-    script = "\n".join(lines)
-    return parse_xir_script(script)
 
+def _get_xir_outputs() -> Iterator[str]:
+    """Returns the names of the supported XIR outputs."""
+    names = ((name, name.lower()) for name in ("Amplitude", "Expval", "Probabilities"))
+    yield from sorted(name for pair in names for name in pair)
```
[the remark this answers is not in the document]
Comment (Mandrenkov) [reply python/jet/interpreter.py:74]: In hindsight, I would agree!

File: python/jet/interpreter.py [hunk 11]
```diff
@@ -43,43 +41,48 @@ def get_xir_library() -> XIRProgram:
...
-        lines.append(line)
+    return program
 
-    script = "\n".join(lines)
-    return parse_xir_script(script)
 
+def _get_xir_outputs() -> Iterator[str]:
+    """Returns the names of the supported XIR outputs."""
+    names = ("Amplitude", "amplitude", "Expval", "expval", "Probabilities", "probabilities")
+    yield from sorted(names)
```
Comment (thisac) [inline python/jet/interpreter.py:74]: I'm being super-picky here (so feel free to ignore), but is it actually even necessary to have `sorted` here?
```suggestion
def _get_xir_outputs() -> Iterator[str]:
    """Returns the names of the supported XIR outputs."""
    names = ("Amplitude", "Expval", "Probabilities", "amplitude", "expval", "probabilities")
    yield from names
```
Or maybe even just have above, skipping the function altogether?
```python
    names = ("Amplitude", "Expval", "Probabilities", "amplitude", "expval", "probabilities")
    for name in names:
        decl = xir.Declaration(type_="out", name=name)
        program.add_declaration(decl)
```

**Review by thisac — APPROVED** [review]
Looks great! 💯

File: python/jet/interpreter.py (hunk 11 above)
Comment (Mandrenkov) [reply python/jet/interpreter.py:74]: Ah, the function is necessary to mock the output names in the tests while the `sorted()` _could_ be removed but would make the order in which the outputs appear in the XIR manifest inconsistent with the order in which the gates appear in the XIR manifest.

**Comment by Mandrenkov** [comment]
Thanks, @thisac and @trbromley!

File: python/jet/interpreter.py (hunk 11 above)
Comment (thisac) [reply python/jet/interpreter.py:74]: Fair enough! Thanks for the explanation! 🙂

**Pull request merged**